### DISCUSSION TITLE
Why my test_loss and test_acc are not improving per epoch

### DISCUSSION THREAD

**Discussion by SailSabnis** [mrdbourke/pytorch-deep-learning#1080] [Q&A]
Hi all. 

Struggling with this one. I am sure this seems pretty basic and would appreciate any help. 
For the FoodVisionMini model using 3 classes (pizza, sushi, steak) example - my train_loss and test_acc seem to remain constant across all epochs. See Screenshot 1 - 
Screenshot 1 - 
<img width="695" alt="Screenshot 2024-09-06 at 6 03 27 AM" src="https://github.com/user-attachments/assets/66c9b14d-0766-43ba-87c4-64f3d02a9ab5">

Here's my train_step and test_step function along with acc function and a train_func that combines both train and test step- 

Screenshot 2 - (Defining TinyVGG model)

<img width="398" alt="Screenshot 2024-09-06 at 6 20 55 AM" src="https://github.com/user-attachments/assets/54417e8c-722f-4414-a954-5debcdd9bf90">

Screenshot 3 - (accuracy function, train_step and test_step function)
<img width="409" alt="Screenshot 2024-09-06 at 6 21 09 AM" src="https://github.com/user-attachments/assets/b0308f92-0f66-4c8d-8391-7da69d8306ef">

Screenshot 4 - (train function combining train_step and test_step and running 5 epochs)
<img width="476" alt="Screenshot 2024-09-06 at 6 21 18 AM" src="https://github.com/user-attachments/assets/1d0b038c-14ea-45de-9a79-d70db9318ba3">

**Comment by SailSabnis** [comment]
@mrdbourke Please don't kill me for tagging you directly but scrambling for help. Scratching my head over this for the last 3 days and couldn't figure what's wrong with my test_loss and test_Acc. Can you (or any other expert if reading this) please help. Many thanks!

**Comment by SailSabnis** [comment]
Hi fellow learners, can someone please help out here? 
<img width="940" alt="Screenshot 2024-09-10 at 2 31 03 PM" src="https://github.com/user-attachments/assets/582ed71d-d9bb-4bfb-befa-ab15f2c74144">

**Comment by mrdbourke** [comment]
Hey @SailSabnis ,

Great question!

Your code looks pretty good to me.

And it looks like your results are similar to what the results are in the example notebook:

```
 0%|          | 0/5 [00:00<?, ?it/s]
Epoch: 1 | train_loss: 1.1078 | train_acc: 0.2578 | test_loss: 1.1360 | test_acc: 0.2604
Epoch: 2 | train_loss: 1.0847 | train_acc: 0.4258 | test_loss: 1.1620 | test_acc: 0.1979
Epoch: 3 | train_loss: 1.1157 | train_acc: 0.2930 | test_loss: 1.1697 | test_acc: 0.1979
Epoch: 4 | train_loss: 1.0956 | train_acc: 0.4141 | test_loss: 1.1384 | test_acc: 0.1979
Epoch: 5 | train_loss: 1.0985 | train_acc: 0.2930 | test_loss: 1.1426 | test_acc: 0.1979
Total training time: 4.935 seconds
```

See here: https://www.learnpytorch.io/04_pytorch_custom_datasets/#77-train-and-evaluate-model-0 

It seems that the TinyVGG model isn't capable of forming a great representation with the current data setup/training setup we have (this is okay, sometimes models don't have the parameters/data to learn, we'll improve results in the Transfer Learning section: https://www.learnpytorch.io/06_pytorch_transfer_learning/).

Your results look similar to the example results too:

<img width="670" alt="Screenshot 2024-09-12 at 2 18 07 PM" src="https://github.com/user-attachments/assets/a0f0e84b-64d4-4b85-b21d-5cc02d612107">

---

If I've misunderstood your question, please feel free to add more information/link to a video/timestamp you're working through so I can see the full code (e.g. what data is being used).

Best, 

Daniel
